### DISCUSSION TITLE
budibase with docker - container "unhealthy"

### DISCUSSION THREAD

**Discussion by diedder** [Budibase/budibase#6714] [General]
I've made a new installation wirh the single container install. Everthything seems to work but docker (and portainer ui) marks the container as "unhealthy"
<img width="1080" alt="Bildschirmfoto 2022-07-14 um 22 27 57" src="https://user-images.githubusercontent.com/25912001/179077804-c6a9bc39-e753-4789-823f-225da3465304.png">
Failurecount raises slowly. How can I fix this? What information to share here for possible investigations?

**Comment by aptkingston** [comment]
Edit: this issue may have been caused by a bug (https://github.com/Budibase/budibase/pull/6767) which we've now fixed. Please try deploying it again with the latest image!

Original reply: Hi @diedder. Did you run it using plain docker (e.g. a "Container" in portainer) or docker compose (a "Stack" in portainer)? I would recommend using docker compose, using the config defined in our docs https://docs.budibase.com/docs/docker#method-2---docker-compose. Running it manually as a container works as well of course, but it's just a lot more error prone as you have to manually ensure the env variables, volumes and ports etc are set up correctly.

Copy and paste that as a portainer stack, change the secrets to something secure and random and then deploy it. I've just ran through that exact procedure and can confirm everything works perfectly.

**Reply by diedder** [reply]
Hi @aptkingston,
I tried with the latest single image but the container is still "unhealty" but seems to work. I used the compose-stack.
For my work I'm now using the multiple container stack.

**Comment by fictionbecomesfact** [comment]
@aptkingston
I ran into the same problem with the latest image.
If you use the 'quick' command below, as described on [Docker Hub](https://hub.docker.com/r/budibase/budibase) and the [Documentation](https://docs.budibase.com/docs/docker), the container is unhealthy and you get the error:
> Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe. Warning: AUTH failed ERROR: Redis is down
```
   docker run -d -t \
   --name=budibase \
   -p 10000:80 \
   -v /local/path/data:/data \
   --restart unless-stopped \
   budibase/budibase:latest
```
When you pass the redis password as an environment variable, the error is solved and the container is healthy:
```
   docker run -d -t \
   --name=budibase \
   -p 10000:80 \
   -v /local/path/data:/data \
   -e REDIS_PASSWORD="budibase" \
   --restart unless-stopped \
   budibase/budibase:latest
```
